### PR TITLE
Don't allow to get pushes for Talk activity

### LINKED ISSUE #6302: "you attended a call" notification shows up again
## Steps to reproduce
1. have a call 
2. see notification
3. dismiss it, it pops up again

image

### Expected behaviour
Do not show it all as I attended it? Or at least do not show it again after dismissing it.

### Actual behaviour
It is shown and even after dismissing it, it pops up again.

## Talk app
on c.nc.c

### PR DESCRIPTION
**Pull request by nickvergessen**
* Chat mentions have custom notifications
* Calls have custom notifications and vanish when you participated
* Invites to new conversations have custom notifications

There are no other activities

Fix #6302

### CODE REVIEW

File: lib/Activity/Setting.php [hunk 1]
```diff
@@ -61,35 +69,11 @@ public function getPriority(): int {
...
-		return true;
-	}
-
-	/**
-	 * @return bool True when the option can be changed for the mail
-	 * @since 11.0.0
-	 */
-	public function canChangeMail(): bool {
-		return true;
+	public function canChangeNotification(): bool {
```
Comment (vitormattos) [inline lib/Activity/Setting.php:72]:
```suggestion
	/**
	 * {@inheritdoc}
	 */
	public function canChangeNotification(): bool {
```

File: lib/Activity/Setting.php [hunk 2]
```diff
@@ -61,35 +69,11 @@ public function getPriority(): int {
...
+		return false;
 	}
 
-	/**
-	 * @return bool True when the option can be changed for the stream
-	 * @since 11.0.0
-	 */
-	public function isDefaultEnabledMail(): bool {
+	public function isDefaultEnabledNotification(): bool {
```
Comment (vitormattos) [inline lib/Activity/Setting.php:76]:
```suggestion
	/**
	 * {@inheritdoc}
	 */
	public function isDefaultEnabledNotification(): bool {
```

File: lib/Activity/Setting.php [hunk 3]
```diff
@@ -51,6 +51,14 @@ public function getName(): string {
 		return $this->l->t('You were invited to a <strong>conversation</strong> or had a <strong>call</strong>');
 	}
 
+	public function getGroupIdentifier(): string {
```
Comment (vitormattos) [inline lib/Activity/Setting.php:54]:
```suggestion
	/**
	 * {@inheritdoc}
	 */
	public function getGroupIdentifier(): string {
```

File: lib/Activity/Setting.php [hunk 4]
```diff
@@ -51,6 +51,14 @@ public function getName(): string {
...
+		return 'other';
+	}
+
+	public function getGroupName() {
```
Comment (vitormattos) [inline lib/Activity/Setting.php:58]:
```suggestion
	/**
	 * {@inheritdoc}
	 */
	public function getGroupName() {
```

**Comment by Matausi29** [comment]
Hey

**Review by vitormattos — APPROVED** [review]
seems ok to me

**Comment by nickvergessen** [comment]
Backport in https://github.com/nextcloud/spreed/pull/6528

**Pull request merged**